src/Range.ts

```diff
@@ -33,18 +33,30 @@ export class Range {
       new Position(endLine, endCharacter)
     )
   }
+
   public static fromNode(node: ts.Node): Range {
-    const sourceFile = node.getSourceFile()
+    // When getting a range of a node a special case is needed for
+    // the constructor case because it does not have a name node.
```

Comment:
1. Thanks for the feedback, I moved this logic and updated the comment. There is an asymmetry between how function declarations and constructor declarations are handled here. For function declarations the identifierNode is passed into `visitSymbolOccurrence` while for constructor declarations its the declaration node itself. This logic (which is now moved) is meant to handle that difference. This logic predates my change.

2. As for the second point on if firstToken can ever be null. I don't think so in this case, however since the type system requires it to be handled (since its nullable) we can either keep the coalescing or throw beforehand. This coalescing behavior was not added as part of my change here but happy to make this cleaner as part of this.